Sources/ArcGISToolkit/Components/Offline/OfflineManager.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+/// An object that maintains state for the offline components.
+@MainActor
+class OfflineManager {
+    /// The shared offline manager.
+    static let `shared` = OfflineManager()
+    
+    /// The action to perform when a job completes.
+    var jobCompletionAction: ((any JobProtocol) -> Void)?
+    
+    /// The job manager used by the offline manager.
+    let jobManager = JobManager.shared
+    
+    /// The jobs managed by this instance.
+    var jobs: [any JobProtocol] { jobManager.jobs }
+    
+    private init() {
+        // Observe each job's status
+        for job in jobManager.jobs {
+            observeJob(job)
+        }
+        
+        // Resume all paused jobs
+        jobManager.resumeAllPausedJobs()
+    }
+    
+    /// Starts a job that will be managed by this instance.
+    /// - Parameter job: The job to start.
+    func start(job: any JobProtocol) {
+        jobManager.jobs.append(job)
+        observeJob(job)
+        job.start()
+    }
+    
+    /// Observes a job for completion.
+    private func observeJob(_ job: any JobProtocol) {
+        Task {
```

Comment:
I think it's ok to have an unstructured task here. It's captures a singleton that remains alive throughout the life of the application. Making this somehow structured doesn't buy us anything that I can think of.

Reply:
As long as the `try? await job.output` async method deterministically returns within finite time, `Task` block should be fine.